activerecord/lib/active_record/relation.rb

```diff
@@ -675,7 +675,13 @@ def values
     end
 
     def inspect
-      entries = records.take([limit_value, 11].compact.min).map!(&:inspect)
+      num_to_take = [limit_value, 11].compact.min
+      if values.any?
+        taken = records.take(num_to_take)
+      else
+        taken = limit(num_to_take).to_a
+      end
+      entries = taken.map!(&:inspect)
```

Comment:
As we're doing an in-place mutation of the array, better to stick to one variable, IMO